### ISSUE TITLE
Video is not resumed on first IR-remote key-press when video is paused + screensaver DIM is active

### ISSUE THREAD

**Issue by guenniguenzelsen** [xbmc/xbmc#24248]
### Describe the bug
Here is a clear and concise description of what the problem is:

If video is paused and the screensaver "DIM" gets activated, the video is not resumed with first IR-remote key-press.
Two key-presses are necessary.
1st IR-key-press = DIM deactivated
2nd IR-key-press = Video is resumed

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

If video is paused and the screensaver "DIM" gets activated, the video shall be resumed with first IR-remote key-press.

Reference:
Kore APP = Video is resumed with single key-press
Yatse 3rd party APP = Video is resumed with single key-press

## Actual Behavior

Paused + dimmed video needs 2 key-presses on the IR remote.

## Possible Fix

Same behavior as Kore-APP or Yatse-APP

### To Reproduce

1. Start video
2. Pause video
3. Wait until the video is dimmed by the screensaver "DIM"
4. Press "Play" on the IR remote
  - 1st press will deactivate DIM
  - 2nd press will resume video

Steps to reproduce expected behavior via Kore-APP
1. Start video
2. Pause video
3. Wait until the video is dimmed by the screensaver "DIM"
4. Press "Play" in Kore-APP
  - 1st press will deactivate DIM and resume video

### Debuglog

The debuglog can be found here:

Video paused + dimmed and resumed by Yatse-APP (expected behaviour)
https://paste.kodi.tv/udewupocin.kodi

### Screenshots 
Here are some links or screenshots to help explain the problem:

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

Based on @DaVukovic analysis, Yatse is sending a "Window Deinit ()" (error-log line 1621) to reach expected behavior

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [x] Linux
 - [ ] macOS
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: LibreELEC-Generic.x86_64-12.0-nightly-20231215-63347fc
 - Kodi version: 21 beta 2, but issue is present since long time

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by DaVukovic** [comment]
>Based on @DaVukovic analysis, Yatse is sending a "Window Deinit ()" (error-log line 1621) to reach expected behavior

Just to answer that. I haven't said that Yatse is sending it. I just noticed in the logs that Yatse is sending the JSON request `Player.PlayPause` and a deinit of a window happens afterwards. As that window doesn't have any name, I suspect it to be the screensaver. 

Hence I guess the JSON-RPC communicates with Kodi on a different level as IR command or keyboard-keypresses do. JSON requests don't seem to be in need to have the screensaver deactivated to be able to interact with Kodi. But that's just an assumption. 

I've also done some additional testings and I am also able to reproduce the issue/problem. 

I've tried with my laptop and with its keyboard. Things I've done: 

* startet kodi
* enabled screensaver
* set it to "Dim"
* set it to 1 minute to jump in
* startet a video
* paused it using the space-bar
* waited until the screensaver jumps in
* hit space bar to resume the video

The only thing which then happens is, that the screensaver gets disabled. Video is still in pause. 

Things which resumes the playback of the video even if the screensaver is enabled:

* `kodi-send -a "Action(PlayPause)"
* JSON-RPC request

Additional tests and "play arounds" I've tried:

wrote a python script:

```
import xbmc
xbmc.executebuiltin(Action("PlayPause"))
```

and mapped that script via `runscript` and a customized "keyboard.xml" to the space bar....still no change. If I hit space while the video is paused AND the screensaver is active, the script doesn't even get triggered.

**Comment by sarbes** [comment]
Waking up on the first button press (without any other action) is the intended behaviour. If remotes behave differently, take it up to them.

**Comment by malvinas2** [comment]
> If remotes behave differently, take it up to them. 

You press the pause button. After a while the screen dims. To continue the video, the pause/start button must be pressed twice. I've known the behavior since I've been using Kodi and always silently assumed this would be 'by design'. In any case, it's not the fault of the remote control.
Was soll denn da die Fernbedienung machen???

**Comment by Jcloak** [comment]
I would also like to request that this issue gets some additional consideration. 

Consider a person that is using Kodi on TV with a normal CEC remote with these 2 scenarios:

**Very short pause scenario**
1) They press pause and get up for 30 seconds to do something
2) The dim screensaver has not yet enabled
3) They press play and the video resumes as expected 

**Longer pause scenario**
1) They press pause and get up for snacks, break, etc. which takes 5-10 minutes
2) The dim screensaver enables
3) They press play and the video does not resume. The dim screensaver goes away
4) They press play again and the video resumes

I understand that it might be "working as intended" to have to press the play button twice, but I personally think it would be a much better user experience if the screensaver goes away and the video resumes with just 1 press of the play button.

Please consider re-opening this to create a more user friendly experience for the average couch Kodi user.

**Comment by neo1973** [comment]
> Please consider re-opening this to create a more user friendly experience for the average couch Kodi user.

I personally prefer the current behaviour.

**Comment by guenniguenzelsen** [comment]
In the actual status we have an inconsistency in the UI

Kodi with official Kodi Kore App: 1 press - immediately resume
Kodi with IR remote: 2 presses

**Comment by Jcloak** [comment]
> > Please consider re-opening this to create a more user friendly experience for the average couch Kodi user.
> 
> I personally prefer the current behaviour.

I'm actually curious why you prefer the current behavior? Are there scenarios where you find it useful to press the play button to make the screensaver go away and not continue watching your video? I think it would be fine if other presses make the screensaver go away without resuming, but I fail to see the benefit of the play button just simply making the screensaver go away and forcing the user to press play again.

**Comment by neo1973** [comment]
> I'm actually curious why you prefer the current behavior?

When the screen is dimmed and the room bright it's not always visible what Kodi is showing on the screen at the moment. I want my first action to brighten the screen and then I can decide what to do. I would find it inconsistent in pressing play in that situation starts the content immediately but pressing the up button only wakes the screen.

**Comment by Jcloak** [comment]
I see. I personally would not press the "Play" button on my remote unless I wanted to resume my content. I also wouldn't forget that I was just watching something on my TV. Thanks for sharing your perspective.

**Comment by thexai** [comment]
It is logical in programming with events: if the screensaver is activated, the first event (any button) deactivates the screensaver and allows the following events to be received/processed.

If the next event is "unpause" then playback continues...

**Comment by Jcloak** [comment]
Yes, I have previously stated that I understand it is "working as intended." I'm suggesting the logic is updated for a better user experience.

**Comment by DaVukovic** [comment]
Just to leave a last commnt from my side as I feel very sorry for @guenniguenzelsen as I encouraged him to open this ticket and now it's simply closed because of personal taste. 

I can mention a bunch of examples where it's the case that a simple button press is totally enough to resume a pause video and if a  screensaver is activated (Prime, Netflix, every standalone player I know of). Only Kodi does it different.

That's generally ok if  Kodi decides to make things different. But that's still bad UX and therefore it's a bug. At least from my POV. That's not personal taste. Because users seeing it behaving interaction different on other apps. 

If I code Kodi in a way that the "stop"-button does "fast-forward" and others complaining that they would expect the video to stop and I answer "Yes, but I did that on purpose and therefore it's not a bug" would we still see that as a UX bug as that's bad UX? Well...I for myself know the answer. The same happened here. The "play button" doesn't resume the video if a screensaver is activated  unless it's either pressed twice or I use a remote app. 

So we agree on that a single button press is "ok" if it's done by an app but not if it's done by an IR remote or by a keyboard. Sounds strange to me, but it is as it is. 

And just to mention it....changing the JSON interface that a double-press is needed would make Kodi UX worse. 

@neo1973 as stated on slack. The official iOS remote app doesn't show the current state of if something is actually playing or not. The remote section simply shows the same buttons and no status of a currently playing video is shown. So the perspective that "I know what Kodi is doing before pressing a button" doesn't count here. 

I've had a long and hard discussion since yesterday with @sarbes to re-open this issue and obviously a few more rules seem to have been added for opening bug tickets. One is, that those UX topics have to be discussed on the Kodi forums first. No other forum is allowed even if it's open readable.  That was new to me and I apologize for missing that new rule. 

I personally won't encourage users anymore to open bug tickets in that regard as I personally don't agree on such a stance. The team should be more open minded to those topics and an open ticket doesn't hurt anyone. Now this is closed and it won't be spoken about anymore. This is the same as "Out of sight, out of mind" and I don't like such a mindset. 

Closing those tickets won't get Kodi any further and at some point developers need to be more user-orientated. 

Jm2c

/me unsubscribes

**Comment by guenniguenzelsen** [comment]
My feeling is that Kodi seems to have good developers, but no product-managers.

The above decision is straight forward from a developer point of view, but far away from how a product manager would approach such topics (e.g. considering use cases, user stories or even just evaluate program statistics, on how many users perform the double-click).

But I'm just a user. I'm using Kodi for free and I'm not contributing anything major. So I'm not in a position to expect anything.

Good luck

**Comment by sarbes** [comment]
I do feel that there are a lot of things mixed up here. I'll try to entangle the different topics.

First of all, the bug report describes exactly how an IR remote should behave in Kodi (from a code point of view). We don't have any design documents that describe how such state changes should take place, so we have to rely on a mix of our source code (as the only form of documentation) and common UX practices. In my eyes, both state changes described in the issue report are valid (to a certain degree, depending on the context), so there is no bug.

The behavior of the mentioned Apps have no relevance to Kodi itself. Any inconsistency is caused by the Apps themselves.  Depending on the context, a sudden resume of playback might be desirable. But overall, I'd say is a shortcoming of those Apps not behaving like a normal remote (if they pose as one). Worthy of note is that the JSON interface those Apps access should provide everything they need to achieve either behavior.

Second, the link to the kodinerds forum has been posted after I've closed the issue. It gives slightly more context. But for non-German speaking developers, it poses a hurdle to participate (even with tools like Google translate). Having it as a base of discussions also excludes a huge part of the user base, as you have to be willing to 1) scan Github constantly to notice the issue, 2) use a translation tool for updates and 3) sign up to a third party website just for a topic like this.

The official forum has far more reach, and the hurdle of participation is way lower. 

Third, a change like this is not straight forward. It is not as black and white as some make it out to be. Just claiming that the current behavior is bogus without presenting sufficient proof is no basis for a bug report. But it is perfectly fine to have a discussion about it.

Fourth, the platform. Issues on GH should be only as long as they need to be. Having a whole discussion about how to interpret current UX rules should not be the content of such an issue. And I do see a huge necessity for such a discussion. 

Similar as written above, the reach of GH is far less than the official forum, where other users and developers can chime in more easily.

Fifth, a few side notes. The described behavior is as old as I can remember. It is also consistent with the behavior when waking up in other UI states. Pretty much every HMI interprets the first input event during such an idle state as a wake-up without any further actions. Having such an animation play out before resuming playback is also good in case the GPU governor downclocked while being idle. And last but not least, the behavior should already be customizable, like so many things in Kodi.

tl;dr: I'd like to encourage every participant to have a discussion in the official forum, where it is visible to way more people. Please provide examples how other devices are managing such input events. If you have design guidelines which describe such behavior (like iOS or Android docs), link them.

On a last note, please don't be discouraged by a closed bug report. Especially for a complex topic like this, further discussion is warranted. As for now, the behavior is by design. But this does not mean that the design can't be changed.

**Comment by sarbes** [comment]
I do feel that there are a lot of things mixed up here. I'll try to entangle the different topics.

First of all, the bug report describes exactly how an IR remote should behave in Kodi (from a code point of view). We don't have any design documents that describe how such state changes should take place, so we have to rely on a mix of our source code (as the only form of documentation) and common UX practices. In my eyes, both state changes described in the issue report are valid (to a certain degree, depending on the context), so there is no bug.

The behavior patterns of the mentioned remote control Apps have no relevance to Kodi itself. Any inconsistency is caused by the Apps themselves. Depending on the context, a sudden resume of playback might be desirable. But overall, I'd say is a shortcoming of those Apps not behaving like a normal remote (if they pose as one). Worthy of note is that the JSON interface those Apps access should provide everything they need to achieve either behavior.

Second, the link to the kodinerds forum has been posted after I've closed the issue. It gives slightly more context. But for non-German speaking developers, it poses a hurdle to participate (even with tools like Google translate). Having it as a base of discussions also excludes a huge part of the user base, as you have to be willing to 1) scan Github constantly to notice the issue, 2) use a translation tool for updates and 3) sign up to a third party website just for a topic like this.

The official forum has far more reach, and the hurdle of participation is way lower.

Third, a change like this is not straight forward. It is not as black and white as some make it out to be. Just claiming that the current behavior is bogus without presenting sufficient proof is no basis for a bug report. But it is perfectly fine to have a discussion about it.

Fourth, the platform. Issues on GH should be only as long as they need to be. Having a whole discussion about how to interpret current UX rules should not be the content of such an issue. And I do see a huge necessity for such a discussion.

Similar as written above, the reach of GH is far less than the official forum, where other users and developers can chime in more easily.

Fifth, a few side notes. The described behavior is as old as I can remember. It is also consistent with the behavior when waking up in other UI states. Pretty much every HMI interprets the first input event during such an idle state as a wake-up without any further actions. Having such an animation play out before resuming playback is also good in case the GPU governor downclocked while being idle. And last but not least, the behavior should already be customizable, like so many things in Kodi.

tl;dr: I'd like to encourage every participant to have a discussion in the official forum, where it is visible to way more people. Please provide examples how other devices are managing such input events. If you have design guidelines which describe such behavior (like iOS or Android docs), link them.

On a last note, please don't be discouraged by a closed bug report. Especially for a complex topic like this, further discussion is warranted. As for now, the behavior is by design. But this does not mean that the design can't be changed.
